### PR TITLE
Calculation of the secondary variables with coupled term

### PR DESCRIPTION
**Pull request by wenqing**
as titled.

Besides, some member names are changed.

### CODE REVIEW

**Comment by wenqing** [comment]
be ready for reviewing once #1715 is merged.

**Comment by wenqing** [comment]
commit ef52004 is just for name changes. If it causes a sort of inconvenience for review, I can put it to another PR.

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h [hunk 1]
```diff
@@ -12,8 +12,11 @@
 
 #pragma once
 
-#include <vector>
+#include <map>
+#include <unordered_map>
 #include <memory>
```
Comment (endJunction) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: I think `map` and `memory` are not needed here. (maybe in the cpp, though)

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h [hunk 2]
```diff
@@ -95,6 +98,11 @@ class LiquidFlowLocalAssembler : public LiquidFlowLocalAssemblerInterface
     void computeSecondaryVariableConcrete(
         double const /*t*/, std::vector<double> const& local_x) override;
 
+    void computeSecondaryVariableWithCoupledProcessConcrete(
+        double const t, std::vector<double> const& local_x,
+        std::unordered_map<std::type_index, const std::vector<double>> const&
+            coupled_local_solutions);
```
Comment (endJunction) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: override specifier missing.

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h (hunk 1 above)
Comment (wenqing) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: Removed (in an amended commit).

File: ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h (hunk 2 above)
Comment (wenqing) [inline ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h]: Added (in an amended commit).

**Pull request merged**